lib/wca_live/scoretaking.ex

```diff
@@ -224,41 +224,61 @@ defmodule WcaLive.Scoretaking do
     if Round.open?(round) do
       {:error, "cannot open this round as it is already open"}
     else
-      Multi.new()
-      |> Multi.run(:finish_previous, fn _repo, _changes ->
-        if round.number > 1 do
-          previous = get_previous_round(round) |> Repo.preload(:results)
-          finish_round(previous)
-        else
-          {:ok, nil}
+      message = validate_previous_rounds(round)
+      if message do
+        {:error, message}
+      else
+        Multi.new()
+        |> Multi.run(:finish_previous, fn _repo, _changes ->
+          if round.number > 1 do
+            previous = get_previous_round(round) |> Repo.preload(:results)
+            finish_round(previous)
+          else
+            {:ok, nil}
+          end
+        end)
+        |> Multi.run(:create_results, fn _repo, _changes ->
+          create_empty_results(round)
+        end)
+        |> Repo.transaction()
+        |> case do
+          {:ok, %{create_results: round}} -> {:ok, round}
+          {:error, _, reason, _} -> {:error, reason}
         end
-      end)
-      |> Multi.run(:create_results, fn _repo, _changes ->
-        create_empty_results(round)
-      end)
-      |> Repo.transaction()
-      |> case do
-        {:ok, %{create_results: round}} -> {:ok, round}
-        {:error, _, reason, _} -> {:error, reason}
       end
     end
   end
 
+  # Check that enough people have competed in the previous round in order to open the next round
+  # See https://www.worldcubeassociation.org/regulations/#9m
+  defp validate_previous_rounds(round) do
+    min = [8, 16, 100]
+    round_results =
+      Enum.to_list(1..round.number - 1 // 1)
+      |> Enum.map(
+        fn round_offset ->
+          Round |> Round.where_sibling(round, -round_offset) |> Repo.one() |> Repo.preload(:results)
+        end
+      )
+      |> Enum.map(fn round -> round.results |> Enum.reject(&Result.empty?/1) |> length() end)
+
+    Enum.find_index(Enum.zip(round_results, min), fn {x, y} -> x < y end)
+    |> case do
+      0 -> "Rounds with 7 or fewer competitors must not have subsequent rounds"
+      1 -> "Rounds with 15 or fewer competitors must have at most one subsequent round"
+      2 -> "Rounds with 99 or fewer competitors must have at most two subsequent rounds"
```

Comment:
Usually we keep `{:error, _}` messages lowercased. We transform to uppercase in other places as needed.

```suggestion
      0 -> "rounds with 7 or fewer competitors must not have subsequent rounds"
      1 -> "rounds with 15 or fewer competitors must have at most one subsequent round"
      2 -> "rounds with 99 or fewer competitors must have at most two subsequent rounds"
```